src/transforms/kubernetes/watch_client.rs

```diff
@@ -0,0 +1,469 @@
+use crate::dns::Resolver;
+use futures::{future::Future, stream::Stream};
+use http::{header, status::StatusCode, uri, Request, Uri};
+use hyper::{client::HttpConnector, Body, Chunk};
+use hyper_openssl::HttpsConnector;
+use k8s_openapi::{
+    api::core::v1::{Pod, WatchPodForAllNamespacesResponse},
+    apimachinery::pkg::apis::meta::v1::WatchEvent,
+    RequestError, Response, ResponseError, WatchOptional,
+};
+use openssl::ssl::{SslConnector, SslConnectorBuilder, SslMethod};
+use snafu::{ResultExt, Snafu};
+
+/// Config which could be loaded from kubeconfig or local kubernetes cluster.
+/// Used to build WatchClient which is in turn used to build Stream of metadata.
+pub struct ClientConfig {
+    resolver: Resolver,
+    token: Option<String>,
+    server: Uri,
+    ssl: Option<SslConnectorBuilder>,
+    node_name: Option<String>,
+}
+
+impl ClientConfig {
+    pub fn build(mut self) -> Result<WatchClient, BuildError> {
```

Comment:
Does this need mut or can it just be `&self`? This would match https://github.com/timberio/vector/blob/master/src/topology/config/mod.rs#L199

Reply:
Needs to be `mut self` since we are consuming `SslConnectorBuilder` which isn't `Clone`.

Reply:
> This would match https://github.com/timberio/vector/blob/master/src/topology/config/mod.rs#L199

Transform which will use this `ClientConfig` will match.

Reply:
Well, I guess this leads me to think maybe we don't want to store the sslbuilder but I'm fine with this for now

Reply:
This will be fixed with using `https_client`